src/main/java/seedu/duke/parsers/DeleteParser.java

```diff
@@ -0,0 +1,36 @@
+package seedu.duke.parsers;
+
+import seedu.duke.commands.Command;
+import seedu.duke.commands.DeleteCommand;
+import seedu.duke.exceptions.ModHappyException;
+
+import java.util.HashMap;
+
+public class DeleteParser extends Parser {
+
+    public static final String MODULE_CODE = "moduleCode";
+    public static final String TASK_NUMBER = "taskNumber";
+    //TODO: make the regex stricter in accepting inputs and extend to deleting tasks from modules
+    private static final String DELETE_FORMAT = "\\s*(\\/t\\s+(?<taskNumber>\\d+))|\\/m\\s+(?<moduleCode>\\w+)";
+
+    public DeleteParser() {
+        super();
+        this.commandFormat = DELETE_FORMAT;
+        groupNames.add(TASK_NUMBER);
+        groupNames.add(MODULE_CODE);
+    }
+
+    @Override
+    public Command parseCommand(String userInput) throws ModHappyException {
+        HashMap<String, String> parsedArguments = parseString(userInput);
+        String taskNumberString = parsedArguments.get(TASK_NUMBER);
+        String moduleCode = parsedArguments.get(MODULE_CODE);
+        if (!moduleCode.isBlank()) {
+            return new DeleteCommand(moduleCode);
+        }
+        if (!taskNumberString.isBlank()) {
+            return new DeleteCommand(Integer.parseInt(taskNumberString));
```

Comment:
Consider wrapping `Integer.parseInt()` in a try-catch block, just in case (although this should not fail during normal code execution because the regex already filters out invalid inputs or negative numbers).